packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClockSection.tsx

```diff
@@ -167,6 +167,8 @@ export const MultiSectionDigitalClockSection = React.forwardRef(
         if (previousActive.current !== activeItem) {
           previousActive.current = activeItem;
         }
+      }
```

Comment:
This used to cause an issue of not changing the focused section when doing the following:
- <kbd>Enter</kbd> on hours section
- <kbd>Shift + Tab</kbd> to return back to hours section
- select a different hour with <kbd>Enter</kbd>
After this, the focused remained on hours section, while it should have been moved to `minutes`.

Reply:
The condition above disturbs me a bit, because if the ref is equal to the active element, we can do `previousActive.current = activeItem` it will modify nothing since it's already this value

```js
if (previousActive.current !== activeItem) {
  previousActive.current = activeItem;
}
```

This code section come from [here](https://github.com/mui/mui-x/commit/575be07ae0e44ebdced1eb17e387c65f0b9e2371)

What about the following diff?

```diff
--- a/packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClockSection.tsx
+++ b/packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClockSection.tsx
@@ -162,16 +162,10 @@ export const MultiSectionDigitalClockSection = React.forwardRef(
       const activeItem = containerRef.current.querySelector<HTMLElement>(
         '[role="option"][tabindex="0"], [role="option"][aria-selected="true"]',
       );
-      if (!activeItem || previousActive.current === activeItem) {
-        // Handle setting the ref to null if the selected item is ever reset via UI
-        if (previousActive.current !== activeItem) {
-          previousActive.current = activeItem;
-        }
-      }
+      previousActive.current = activeItem;
       if (!activeItem) {
         return;
       }
-      previousActive.current = activeItem;
       if (active && autoFocus) {
         activeItem.focus();
       }

```

Reply:
Applying your diff reverts this fix: https://github.com/mui/mui-x/issues/9389.
I'll look if there is another way to simplify the code. 👌

Reply:
@alexfauquette WDYT about https://github.com/mui/mui-x/pull/11419/commits/185c222643d4151fba380985b5b6aafcd7bfa1bd?
It seems like a decent compromise and a tiny "perf improvement".
I did not see a regression with this given simplification, the linked issue is still resolved. 🤔 
Could you think of anything that could have changed for the worse? 🤔

Reply:
I've made another small update: https://github.com/mui/mui-x/pull/11419/commits/03a31f90cd0737351ff20e13efacc569cd7cf9ba to avoid the problem of the relevant section not being focused when reselecting a previous section.